cli/src/main.rs

```diff
@@ -181,11 +182,11 @@ fn build(args: BuildArgs) -> ZResult<()> {
 }
 
 pub async fn serve(args: ServeArgs) -> ZResult<()> {
-    let mut proj = Project::new(args.compile)?;
+    let proj = std::sync::Mutex::new(Project::new(args.compile.clone())?);
 
     // Build the book if it hasn't been built yet
     if !args.no_build {
-        proj.build()?;
+        proj.lock().expect("Cannot get lock").build()?;
 
         // since we don't need the compilation cache anymore, we can evict it
         comemo::evict(0);
```

Comment:
the compilation cache should be evicted correctly. when watching, we don't evict it totally after first compilation and call `evict(N)` after later compilations. iirc `N` is `10` in `typst-cli`.